sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/error/NewAnalysisError.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.batch.sensor.error;
+
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.TextPointer;
+import org.sonar.api.batch.sensor.Sensor;
+
+/**
+ * Represents an analysis error, such as a parsing error, that occurs during the execution of a {@link Sensor}.
+ * 
+ * @since 6.0
+ */
+public interface NewAnalysisError {
+  /**
+   * The file that was being processed when the error occurred. This field must be set before saving the error.
+   */
+  NewAnalysisError onFile(InputFile inputFile);
+
+  /**
+   * Message about the error. This field is optional.
+   */
+  NewAnalysisError message(String message);
+
+  /**
+   * Location of this error. This field is optional.
+   */
+  NewAnalysisError at(TextPointer location);
+
+  /**
+   * Save this error information.
```

Comment:
Maybe we could document that:
  * it is supported to save an analysis error several times for the same file, but behavior is not specified (ie do we display all errors, or only the first/last one)
  * it is supported to report issues and analysis error for the same file, but again result is not specified (do we display issues or not?)